File: src/main/java/miretz/ycloud/services/DatabaseService.java
package miretz.ycloud.services;

import java.util.List;

import miretz.ycloud.models.Document;

public interface DatabaseService {

	/** user interaction */

	boolean checkUserPassword(String username, String password);

	void addUser(String username, String password);

	void removeUser(String username);

	List<String> listUsernames();

	/** file interaction */

	void addDocument(Document document);

	List<Document> getAllDocuments();

	List<Document> getDescendants(String parentId);

	Document getDocument(String contentId);

	Document findByMetadataValue(String key, String value);

	void deleteDocument(String contentId);

	void deleteAllDocuments();

}

File: src/main/java/miretz/ycloud/services/DatabaseService.kt
package miretz.ycloud.services


import miretz.ycloud.models.Document

interface DatabaseService {

    /** user interaction  */

    fun checkUserPassword(username: String, password: String): Boolean

    fun addUser(username: String, password: String)

    fun removeUser(username: String)

    fun listUsernames(): List<String>

    /** file interaction  */

    fun addDocument(documentToStore: Document)

    fun getAllDocuments(): List<Document>

    fun getDescendants(parentId: String): List<Document>

    fun findDocument(contentId: String): Document

    fun findByMetadataValue(key: String, value: String): Document

    fun deleteDocument(contentId: String)

    fun deleteAllDocuments()

}
